skypy/utils/tests/test_random.py

```diff
@@ -34,3 +36,32 @@ def calc_cdf(x):
     # Test the distribution of galaxy properties follows the right distribution
     p_value = scipy.stats.kstest(sample, calc_cdf)[1]
     assert p_value > 0.01
+
+
+def test_conditional_prob_shmr():
+
+    # Test the conditional_prob_shmr function against precomputed values
+    x = np.linspace(11., 15., 20)
+    cdf = random.conditional_prob_shmr(min(x), max(x), 20)
+    sample = np.array([
+             0., 0.05263158, 0.10526316, 0.15789474, 0.21052632,
+             0.26315789, 0.31578947, 0.36842105, 0.42105263, 0.47368421,
+             0.52631579, 0.57894737, 0.63157895, 0.68421053, 0.73684211,
+             0.78947368, 0.84210526, 0.89473684, 0.94736842, 1.])
+
+    np.testing.assert_allclose(cdf, sample)
+
+    # Test the conditional_prob_shmr function sampling dimensionless x values
+    x_min = 11.
+    x_max = 15.
+
+    def calc_cdf(x):
+        x = np.linspace(x_min, x_max, 1000)
+        cdf = integrate.cumtrapz(sp.erf(x), x, initial=0)
+        cdf = cdf/cdf[-1]
+        return cdf
+    sample = random.conditional_prob_shmr(x_min=x_min, x_max=x_max, size=1000)
+
+    # Test the distribution of galaxy properties follows the right distribution
+    p_value = scipy.stats.kstest(sample, calc_cdf)[1]
+    assert p_value > 0.01
```

Comment:
Looks good!

Reply:
This test is not working as intended. `kstest` takes samples and a cdf as arguments to compare. But you are passing it the same cdf twice since `conditional_prob_shmr` returns the cdf, not samples. The test appears to pass because the kstest calculates p_value = 1 but this is clearly incorrect.